audformat/core/utils.py

```diff
@@ -212,6 +212,64 @@ def concat(
         return df
 
 
+def duration(
+        obj: typing.Union[pd.Index, pd.Series, pd.DataFrame],
+        *,
+        num_workers: int = 1,
+        verbose: bool = False,
+) -> pd.Timedelta:
+    r"""Total duration of all entries present in the object.
+
+    The object might contain a segmented or a filewise index.
+    For a segmented index the duration is calculated
+    from its start and end values.
+    If an end value is ``NaT``
+    or the object contains a filewise index
+    the duration is calculated from the media file
+    by calling :func:`audiofile.duration`.
+
+    Args:
+        obj: object conform to
+            :ref:`table specifications <data-tables:Tables>`
+        num_workers: number of parallel jobs.
+            Only relevant when the duration of the files
+            needs to be detected from the file.
+            If ``None`` will be set to the number of processors
+            on the machine multiplied by 5
+        verbose: show progress bar when duration needs to be detected
+
+    Returns:
+        duration
+
+    Example:
+
+        >>> idx = segmented_index(
+        ...     files=['a', 'b', 'c'],
+        ...     starts=[0, 1, 3],
+        ...     ends=[1, 2, 4],
+        ... )
+        >>> duration(idx)
+        Timedelta('0 days 00:00:03')
+
+    """
+    obj = to_segmented_index(
```

Comment:
`to_segmented_index()` also has a `root` argument. I think we should also support it in this function.

Reply:
Good point, I added the `root` argument and pass it on.